---
title: "Abundance Tables"
output:
  html_document:
    keep_md: true
---

# Create Abundance Tables
Description.

KEGG file ko_genes.list not available due to licensing. Outputs from the script blast2tsv.pl can be found in the intermediate_results directory.

Ion Torrent Viral Metagenome.

```{r, engine='bash'}
# perl scripts/blast2tsv.pl -blast_file=intermediate_results/vmg.ublast.kegg.txt -bowtie_dir=vmg_orfs_bowtie -gene_ko=ko_genes.list
# 
# mkdir viral_blast2tsv
# mv *no_ko_hit.txt viral_blast2tsv/
# mv *_abundances.txt viral_blast2tsv/
# mv *_abundance.txt viral_blast2tsv/
# mv blast_top_hit.txt viral_blast2tsv/
# mv orf_ko_assignment.txt viral_blast2tsv/
```

Merge all the normalized abundance tables together.

```{r, engine='bash'}
mv intermediate_results/viral_blast2tsv ./
for f in viral_blast2tsv/*orf_abundance.txt
do
	filename=$(basename "$f")
	filename="${filename%.txt}"
	biom convert --table-type="OTU table" --to-json -i $f -o viral_blast2tsv/$filename.biom
done

merge_otu_tables.py -i viral_blast2tsv/VMG.1_trimm.orf_abundance.biom,viral_blast2tsv/VMG.2_trimm.orf_abundance.biom,viral_blast2tsv/VMG.3_trimm.orf_abundance.biom,viral_blast2tsv/VMG.4_trimm.orf_abundance.biom,viral_blast2tsv/VMG.5_trimm.orf_abundance.biom,viral_blast2tsv/VMG.6_trimm.orf_abundance.biom,viral_blast2tsv/VMG.7_trimm.orf_abundance.biom,viral_blast2tsv/VMG.8_trimm.orf_abundance.biom,viral_blast2tsv/VMG.9_trimm.orf_abundance.biom,viral_blast2tsv/VMG.10_trimm.orf_abundance.biom,viral_blast2tsv/VMG.11_trimm.orf_abundance.biom,viral_blast2tsv/VMG.12_trimm.orf_abundance.biom,viral_blast2tsv/VMG.13_trimm.orf_abundance.biom,viral_blast2tsv/VMG.14_trimm.orf_abundance.biom,viral_blast2tsv/VMG.15_trimm.orf_abundance.biom -o viral_blast2tsv/VMG.orf_abundance.biom

biom summarize-table -i viral_blast2tsv/VMG.orf_abundance.biom -o viral_blast2tsv/VMG.orf_abundance_summarize.txt
	
biom convert -i viral_blast2tsv/VMG.orf_abundance.biom -o viral_blast2tsv/VMG.orf_abundance.txt --table-type="OTU table" --to-tsv
```

Ion Torrent Total Metagenome.

```{r, engine='bash'}
# perl scripts/blast2tsv.pl -blast_file=intermediate_results/bmg.ublast.kegg.txt -bowtie_dir=bmg_orfs_bowtie -gene_ko=ko_genes.list
# 
# mkdir bmg_blast2tsv
# mv *no_ko_hit.txt bmg_blast2tsv/
# mv *_abundances.txt bmg_blast2tsv/
# mv *_abundance.txt bmg_blast2tsv/
# mv blast_top_hit.txt bmg_blast2tsv/
# mv orf_ko_assignment.txt bmg_blast2tsv/
```

Merge all the normalized abundance tables together.

```{r, engine='bash'}
mv intermediate_results/bmg_blast2tsv ./
for f in bmg_blast2tsv/*orf_abundance.txt
do
  filename=$(basename "$f")
	filename="${filename%.txt}"
	biom convert --table-type="OTU table" --to-json -i $f -o bmg_blast2tsv/$filename.biom
done

merge_otu_tables.py -i bmg_blast2tsv/BMG.1.orf_abundance.biom,bmg_blast2tsv/BMG.2.orf_abundance.biom,bmg_blast2tsv/BMG.3.orf_abundance.biom,bmg_blast2tsv/BMG.4.orf_abundance.biom,bmg_blast2tsv/BMG.5.orf_abundance.biom,bmg_blast2tsv/BMG.6.orf_abundance.biom,bmg_blast2tsv/BMG.7.orf_abundance.biom,bmg_blast2tsv/BMG.8.orf_abundance.biom,bmg_blast2tsv/BMG.9.orf_abundance.biom,bmg_blast2tsv/BMG.10.orf_abundance.biom,bmg_blast2tsv/BMG.11.orf_abundance.biom,bmg_blast2tsv/BMG.12.orf_abundance.biom,bmg_blast2tsv/BMG.13.orf_abundance.biom,bmg_blast2tsv/BMG.14.orf_abundance.biom,bmg_blast2tsv/BMG.15.orf_abundance.biom,bmg_blast2tsv/BMG.16.orf_abundance.biom,bmg_blast2tsv/BMG.17.orf_abundance.biom,bmg_blast2tsv/BMG.18.orf_abundance.biom,bmg_blast2tsv/BMG.19.orf_abundance.biom,bmg_blast2tsv/BMG.20.orf_abundance.biom -o bmg_blast2tsv/BMG.orf_abundance.biom

biom summarize-table -i bmg_blast2tsv/BMG.orf_abundance.biom -o bmg_blast2tsv/BMG.orf_abundance_summarize.txt
	
biom convert -i bmg_blast2tsv/BMG.orf_abundance.biom -o bmg_blast2tsv/BMG.orf_abundance.txt --table-type="OTU table" --to-tsv
```

Illumina MiSeq Viral Metagenome.

```{r, engine='bash'}
# perl scripts/blast2tsv.pl -blast_file=intermediate_results/vmg.illumina.ublast.kegg.txt -bowtie_dir=vmg_illumina_orfs_bowtie -gene_ko=ko_genes.list
# 
# mkdir viral_illumina_blast2tsv
# mv *no_ko_hit.txt viral_illumina_blast2tsv
# mv *_abundances.txt viral_illumina_blast2tsv
# mv *_abundance.txt viral_illumina_blast2tsv
# mv blast_top_hit.txt viral_illumina_blast2tsv
# mv orf_ko_assignment.txt viral_illumina_blast2tsv
```

Merge all the normalized abundance tables together.

```{r, engine='bash'}
mv intermediate_results/bmg_blast2tsv ./
for f in viral_illumina_blast2tsv/*orf_abundance.txt
do
  filename=$(basename "$f")
	filename="${filename%.txt}"
	biom convert --table-type="OTU table" --to-json -i $f -o viral_illumina_blast2tsv/$filename.biom
done

merge_otu_tables.py -i viral_illumina_blast2tsv/V4.illumina.cat.fastq.orf_abundance.biom,viral_illumina_blast2tsv/V9.illumina.cat.fastq.orf_abundance.biom,viral_illumina_blast2tsv/V10.illumina.cat.fastq.orf_abundance.biom,viral_illumina_blast2tsv/V11.illumina.cat.fastq.orf_abundance.biom,viral_illumina_blast2tsv/V12.illumina.cat.fastq.orf_abundance.biom,viral_illumina_blast2tsv/V13.illumina.cat.fastq.orf_abundance.biom -o viral_illumina_blast2tsv/VMG.illumina.orf_abundance.biom

biom summarize-table -i viral_illumina_blast2tsv/VMG.illumina.orf_abundance.biom  -o viral_illumina_blast2tsv/VMG.illumina.orf_abundance_summarize.txt
	
biom convert -i viral_illumina_blast2tsv/VMG.illumina.orf_abundance.biom -o viral_illumina_blast2tsv/VMG.illumina.orf_abundance.txt --table-type="OTU table" --to-tsv
```