---
title: "Abundance Tables"
output:
  html_document:
    keep_md: true
---

Take the results of searching ORFs against the KEGG database and the results from mapping reads to ORFs and obtain abundances of ORFs and KOs. KEGG file 'ko_genes.list', which maps KEGG genes from this version of KEGG to their respective KO families, can not be provided due to licensing. Output abundance tables can be found in the intermediate_results directory though.

## Viral Metagenomes

```{r, engine='bash'}
python scripts/kegg_blast2tsv.py --blast=intermediate_results/vmg.ublast.kegg.txt --bowtie_dir=vmg_orfs_bowtie --gene_ko_mapping=kegg/genes/ko/ko_genes.list

mkdir viral_blast2tsv
mv *orf_*_abundance.txt viral_blast2tsv
rm *ko_*_abundance.txt

for f in viral_blast2tsv/*orf_norm_abundance.txt
do
	filename=$(basename "$f")
	filename="${filename%.txt}"
	biom convert --table-type="OTU table" --to-json -i $f -o viral_blast2tsv/$filename.biom
done

merge_otu_tables.py -i viral_blast2tsv/VMG.1_trimm.orf_norm_abundance.biom,viral_blast2tsv/VMG.2_trimm.orf_norm_abundance.biom,viral_blast2tsv/VMG.3_trimm.orf_norm_abundance.biom,viral_blast2tsv/VMG.4_trimm.orf_norm_abundance.biom,viral_blast2tsv/VMG.5_trimm.orf_norm_abundance.biom,viral_blast2tsv/VMG.6_trimm.orf_norm_abundance.biom,viral_blast2tsv/VMG.7_trimm.orf_norm_abundance.biom,viral_blast2tsv/VMG.8_trimm.orf_norm_abundance.biom,viral_blast2tsv/VMG.9_trimm.orf_norm_abundance.biom,viral_blast2tsv/VMG.10_trimm.orf_norm_abundance.biom,viral_blast2tsv/VMG.11_trimm.orf_norm_abundance.biom,viral_blast2tsv/VMG.12_trimm.orf_norm_abundance.biom,viral_blast2tsv/VMG.13_trimm.orf_norm_abundance.biom,viral_blast2tsv/VMG.14_trimm.orf_norm_abundance.biom,viral_blast2tsv/VMG.15_trimm.orf_norm_abundance.biom -o viral_blast2tsv/VMG.orf_norm_abundance.biom

biom summarize-table -i viral_blast2tsv/VMG.orf_norm_abundance.biom -o viral_blast2tsv/VMG.orf_norm_abundance_summarize.txt
	
biom convert -i viral_blast2tsv/VMG.orf_norm_abundance.biom -o viral_blast2tsv/VMG.orf_norm_abundance.txt --table-type="OTU table" --to-tsv
```

Filter viral KOs for viral homology using the results from searching ORFs against the PHAST Prophage and Virus Database. The script below generates new KO abundance tables that originate from ORFs that were on a contig that contained another ORF with a significant hit to the viral databse.

```{r, engine='bash'}
python scripts/viral_ko_filter.py -v vmg.ublast.phast.txt -k intermediate_results/vmg.ublast.kegg.txt 
mv viral_filtered_kegg_blast.txt vmg.ublast.kegg.filter.txt

python scripts/kegg_blast2tsv.py --blast=vmg.ublast.kegg.filter.txt --bowtie_dir=vmg_orfs_bowtie --gene_ko_mapping=kegg/genes/ko/ko_genes.list

rm *orf_*_abundance.txt
mv *ko_*_abundance.txt viral_blast2tsv

for f in viral_blast2tsv/*ko_norm_abundance.txt
do
	filename=$(basename "$f")
	filename="${filename%.txt}"
	biom convert --table-type="OTU table" --to-json -i $f -o viral_blast2tsv/$filename.biom
done

merge_otu_tables.py -i viral_blast2tsv/VMG.1_trimm.ko_norm_abundance.biom,viral_blast2tsv/VMG.2_trimm.ko_norm_abundance.biom,viral_blast2tsv/VMG.3_trimm.ko_norm_abundance.biom,viral_blast2tsv/VMG.4_trimm.ko_norm_abundance.biom,viral_blast2tsv/VMG.5_trimm.ko_norm_abundance.biom,viral_blast2tsv/VMG.6_trimm.ko_norm_abundance.biom,viral_blast2tsv/VMG.7_trimm.ko_norm_abundance.biom,viral_blast2tsv/VMG.8_trimm.ko_norm_abundance.biom,viral_blast2tsv/VMG.9_trimm.ko_norm_abundance.biom,viral_blast2tsv/VMG.10_trimm.ko_norm_abundance.biom,viral_blast2tsv/VMG.11_trimm.ko_norm_abundance.biom,viral_blast2tsv/VMG.12_trimm.ko_norm_abundance.biom,viral_blast2tsv/VMG.13_trimm.ko_norm_abundance.biom,viral_blast2tsv/VMG.14_trimm.ko_norm_abundance.biom,viral_blast2tsv/VMG.15_trimm.ko_norm_abundance.biom -o viral_blast2tsv/VMG.ko_norm_abundance.biom

biom summarize-table -i viral_blast2tsv/VMG.ko_norm_abundance.biom -o viral_blast2tsv/VMG.ko_norm_abundance_summarize.txt
	
biom convert -i viral_blast2tsv/VMG.ko_norm_abundance.biom -o viral_blast2tsv/VMG.ko_norm_abundance.txt --table-type="OTU table" --to-tsv

for f in viral_blast2tsv/*ko_raw_abundance.txt
do
	filename=$(basename "$f")
	filename="${filename%.txt}"
	biom convert --table-type="OTU table" --to-json -i $f -o viral_blast2tsv/$filename.biom
done

merge_otu_tables.py -i viral_blast2tsv/VMG.1_trimm.ko_raw_abundance.biom,viral_blast2tsv/VMG.2_trimm.ko_raw_abundance.biom,viral_blast2tsv/VMG.3_trimm.ko_raw_abundance.biom,viral_blast2tsv/VMG.4_trimm.ko_raw_abundance.biom,viral_blast2tsv/VMG.5_trimm.ko_raw_abundance.biom,viral_blast2tsv/VMG.6_trimm.ko_raw_abundance.biom,viral_blast2tsv/VMG.7_trimm.ko_raw_abundance.biom,viral_blast2tsv/VMG.8_trimm.ko_raw_abundance.biom,viral_blast2tsv/VMG.9_trimm.ko_raw_abundance.biom,viral_blast2tsv/VMG.10_trimm.ko_raw_abundance.biom,viral_blast2tsv/VMG.11_trimm.ko_raw_abundance.biom,viral_blast2tsv/VMG.12_trimm.ko_raw_abundance.biom,viral_blast2tsv/VMG.13_trimm.ko_raw_abundance.biom,viral_blast2tsv/VMG.14_trimm.ko_raw_abundance.biom,viral_blast2tsv/VMG.15_trimm.ko_raw_abundance.biom -o viral_blast2tsv/VMG.ko_raw_abundance.biom

biom summarize-table -i viral_blast2tsv/VMG.ko_raw_abundance.biom -o viral_blast2tsv/VMG.ko_raw_abundance_summarize.txt
	
biom convert -i viral_blast2tsv/VMG.ko_raw_abundance.biom -o viral_blast2tsv/VMG.ko_raw_abundance.txt --table-type="OTU table" --to-tsv
```

## Microbial Metagenomes

```{r, engine='bash'}
python scripts/kegg_blast2tsv.py --blast=intermediate_results/bmg.ublast.kegg.txt --bowtie_dir=bmg_orfs_bowtie --gene_ko_mapping=kegg/genes/ko/ko_genes.list

mkdir bmg_blast2tsv
mv *orf_*_abundance.txt bmg_blast2tsv
mv *ko_*_abundance.txt bmg_blast2tsv

for f in bmg_blast2tsv/*orf_norm_abundance.txt
do
  filename=$(basename "$f")
	filename="${filename%.txt}"
	biom convert --table-type="OTU table" --to-json -i $f -o bmg_blast2tsv/$filename.biom
done

merge_otu_tables.py -i bmg_blast2tsv/BMG.1.orf_norm_abundance.biom,bmg_blast2tsv/BMG.2.orf_norm_abundance.biom,bmg_blast2tsv/BMG.3.orf_norm_abundance.biom,bmg_blast2tsv/BMG.4.orf_norm_abundance.biom,bmg_blast2tsv/BMG.5.orf_norm_abundance.biom,bmg_blast2tsv/BMG.6.orf_norm_abundance.biom,bmg_blast2tsv/BMG.7.orf_norm_abundance.biom,bmg_blast2tsv/BMG.8.orf_norm_abundance.biom,bmg_blast2tsv/BMG.9.orf_norm_abundance.biom,bmg_blast2tsv/BMG.10.orf_norm_abundance.biom,bmg_blast2tsv/BMG.11.orf_norm_abundance.biom,bmg_blast2tsv/BMG.12.orf_norm_abundance.biom,bmg_blast2tsv/BMG.13.orf_norm_abundance.biom,bmg_blast2tsv/BMG.14.orf_norm_abundance.biom,bmg_blast2tsv/BMG.15.orf_norm_abundance.biom,bmg_blast2tsv/BMG.16.orf_norm_abundance.biom,bmg_blast2tsv/BMG.17.orf_norm_abundance.biom,bmg_blast2tsv/BMG.18.orf_norm_abundance.biom,bmg_blast2tsv/BMG.19.orf_norm_abundance.biom,bmg_blast2tsv/BMG.20.orf_norm_abundance.biom -o bmg_blast2tsv/BMG.orf_norm_abundance.biom

biom summarize-table -i bmg_blast2tsv/BMG.orf_norm_abundance.biom -o bmg_blast2tsv/BMG.orf_norm_abundance_summarize.txt
	
biom convert -i bmg_blast2tsv/BMG.orf_norm_abundance.biom -o bmg_blast2tsv/BMG.orf_norm_abundance.txt --table-type="OTU table" --to-tsv


for f in bmg_blast2tsv/*ko_norm_abundance.txt
do
  filename=$(basename "$f")
	filename="${filename%.txt}"
	biom convert --table-type="OTU table" --to-json -i $f -o bmg_blast2tsv/$filename.biom
done

merge_otu_tables.py -i bmg_blast2tsv/BMG.1.ko_norm_abundance.biom,bmg_blast2tsv/BMG.2.ko_norm_abundance.biom,bmg_blast2tsv/BMG.3.ko_norm_abundance.biom,bmg_blast2tsv/BMG.4.ko_norm_abundance.biom,bmg_blast2tsv/BMG.5.ko_norm_abundance.biom,bmg_blast2tsv/BMG.6.ko_norm_abundance.biom,bmg_blast2tsv/BMG.7.ko_norm_abundance.biom,bmg_blast2tsv/BMG.8.ko_norm_abundance.biom,bmg_blast2tsv/BMG.9.ko_norm_abundance.biom,bmg_blast2tsv/BMG.10.ko_norm_abundance.biom,bmg_blast2tsv/BMG.11.ko_norm_abundance.biom,bmg_blast2tsv/BMG.12.ko_norm_abundance.biom,bmg_blast2tsv/BMG.13.ko_norm_abundance.biom,bmg_blast2tsv/BMG.14.ko_norm_abundance.biom,bmg_blast2tsv/BMG.15.ko_norm_abundance.biom,bmg_blast2tsv/BMG.16.ko_norm_abundance.biom,bmg_blast2tsv/BMG.17.ko_norm_abundance.biom,bmg_blast2tsv/BMG.18.ko_norm_abundance.biom,bmg_blast2tsv/BMG.19.ko_norm_abundance.biom,bmg_blast2tsv/BMG.20.ko_norm_abundance.biom -o bmg_blast2tsv/BMG.ko_norm_abundance.biom

biom summarize-table -i bmg_blast2tsv/BMG.ko_norm_abundance.biom -o bmg_blast2tsv/BMG.ko_norm_abundance_summarize.txt
	
biom convert -i bmg_blast2tsv/BMG.ko_norm_abundance.biom -o bmg_blast2tsv/BMG.ko_norm_abundance.txt --table-type="OTU table" --to-tsv

for f in bmg_blast2tsv/*ko_raw_abundance.txt
do
  filename=$(basename "$f")
	filename="${filename%.txt}"
	biom convert --table-type="OTU table" --to-json -i $f -o bmg_blast2tsv/$filename.biom
done

merge_otu_tables.py -i bmg_blast2tsv/BMG.1.ko_raw_abundance.biom,bmg_blast2tsv/BMG.2.ko_raw_abundance.biom,bmg_blast2tsv/BMG.3.ko_raw_abundance.biom,bmg_blast2tsv/BMG.4.ko_raw_abundance.biom,bmg_blast2tsv/BMG.5.ko_raw_abundance.biom,bmg_blast2tsv/BMG.6.ko_raw_abundance.biom,bmg_blast2tsv/BMG.7.ko_raw_abundance.biom,bmg_blast2tsv/BMG.8.ko_raw_abundance.biom,bmg_blast2tsv/BMG.9.ko_raw_abundance.biom,bmg_blast2tsv/BMG.10.ko_raw_abundance.biom,bmg_blast2tsv/BMG.11.ko_raw_abundance.biom,bmg_blast2tsv/BMG.12.ko_raw_abundance.biom,bmg_blast2tsv/BMG.13.ko_raw_abundance.biom,bmg_blast2tsv/BMG.14.ko_raw_abundance.biom,bmg_blast2tsv/BMG.15.ko_raw_abundance.biom,bmg_blast2tsv/BMG.16.ko_raw_abundance.biom,bmg_blast2tsv/BMG.17.ko_raw_abundance.biom,bmg_blast2tsv/BMG.18.ko_raw_abundance.biom,bmg_blast2tsv/BMG.19.ko_raw_abundance.biom,bmg_blast2tsv/BMG.20.ko_raw_abundance.biom -o bmg_blast2tsv/BMG.ko_raw_abundance.biom

biom summarize-table -i bmg_blast2tsv/BMG.ko_raw_abundance.biom -o bmg_blast2tsv/BMG.ko_raw_abundance_summarize.txt
	
biom convert -i bmg_blast2tsv/BMG.ko_raw_abundance.biom -o bmg_blast2tsv/BMG.ko_raw_abundance.txt --table-type="OTU table" --to-tsv
```

## Deep Viral Metagenomes

```{r, engine='bash'}
python scripts/kegg_blast2tsv.py --blast=intermediate_results/vmg.illumina.ublast.kegg.txt --bowtie_dir=vmg_illumina_orfs_bowtie --gene_ko_mapping=kegg/genes/ko/ko_genes.list

mkdir viral_illumina_blast2tsv
mv *orf_*_abundance.txt viral_illumina_blast2tsv
rm *ko_*_abundance.txt

for f in viral_illumina_blast2tsv/*orf_norm_abundance.txt
do
  filename=$(basename "$f")
	filename="${filename%.txt}"
	biom convert --table-type="OTU table" --to-json -i $f -o viral_illumina_blast2tsv/$filename.biom
done

merge_otu_tables.py -i viral_illumina_blast2tsv/V4.illumina.cat.fastq.orf_norm_abundance.biom,viral_illumina_blast2tsv/V9.illumina.cat.fastq.orf_norm_abundance.biom,viral_illumina_blast2tsv/V10.illumina.cat.fastq.orf_norm_abundance.biom,viral_illumina_blast2tsv/V11.illumina.cat.fastq.orf_norm_abundance.biom,viral_illumina_blast2tsv/V12.illumina.cat.fastq.orf_norm_abundance.biom,viral_illumina_blast2tsv/V13.illumina.cat.fastq.orf_norm_abundance.biom -o viral_illumina_blast2tsv/VMG.illumina.orf_norm_abundance.biom

biom summarize-table -i viral_illumina_blast2tsv/VMG.illumina.orf_norm_abundance.biom  -o viral_illumina_blast2tsv/VMG.illumina.orf_norm_abundance_summarize.txt
	
biom convert -i viral_illumina_blast2tsv/VMG.illumina.orf_norm_abundance.biom -o viral_illumina_blast2tsv/VMG.illumina.orf_norm_abundance.txt --table-type="OTU table" --to-tsv
```

Filter viral KOs for viral homology using the results from searching ORFs against the PHAST Prophage and Virus Database. The script below filter KOs that originate from ORFs that were on a contig that contained another ORF with a significant hit to the viral databse.

```{r, engine='bash'}
python scripts/viral_ko_filter.py -v vmg.illumina.ublast.phast.txt -k intermediate_results/vmg.illumina.ublast.kegg.txt 
mv viral_filtered_kegg_blast.txt vmg.illumina.ublast.kegg.filter.txt

python scripts/kegg_blast2tsv.py --blast=vmg.illumina.ublast.kegg.filter.txt --bowtie_dir=vmg_illumina_orfs_bowtie --gene_ko_mapping=kegg/genes/ko/ko_genes.list

rm *orf_*_abundance.txt
mv *ko_*_abundance.txt viral_illumina_blast2tsv

for f in viral_illumina_blast2tsv/*ko_norm_abundance.txt
do
	filename=$(basename "$f")
	filename="${filename%.txt}"
	biom convert --table-type="OTU table" --to-json -i $f -o viral_illumina_blast2tsv/$filename.biom
done

merge_otu_tables.py -i viral_illumina_blast2tsv/V4.illumina.cat.fastq.ko_norm_abundance.biom,viral_illumina_blast2tsv/V9.illumina.cat.fastq.ko_norm_abundance.biom,viral_illumina_blast2tsv/V10.illumina.cat.fastq.ko_norm_abundance.biom,viral_illumina_blast2tsv/V11.illumina.cat.fastq.ko_norm_abundance.biom,viral_illumina_blast2tsv/V12.illumina.cat.fastq.ko_norm_abundance.biom,viral_illumina_blast2tsv/V13.illumina.cat.fastq.ko_norm_abundance.biom -o viral_illumina_blast2tsv/VMG.illumina.ko_norm_abundance.biom

biom summarize-table -i viral_illumina_blast2tsv/VMG.illumina.ko_norm_abundance.biom  -o viral_illumina_blast2tsv/VMG.illumina.ko_norm_abundance_summarize.txt
	
biom convert -i viral_illumina_blast2tsv/VMG.illumina.ko_norm_abundance.biom -o viral_illumina_blast2tsv/VMG.illumina.ko_norm_abundance.txt --table-type="OTU table" --to-tsv

for f in viral_illumina_blast2tsv/*ko_raw_abundance.txt
do
	filename=$(basename "$f")
	filename="${filename%.txt}"
	biom convert --table-type="OTU table" --to-json -i $f -o viral_illumina_blast2tsv/$filename.biom
done

merge_otu_tables.py -i viral_illumina_blast2tsv/V4.illumina.cat.fastq.ko_raw_abundance.biom,viral_illumina_blast2tsv/V9.illumina.cat.fastq.ko_raw_abundance.biom,viral_illumina_blast2tsv/V10.illumina.cat.fastq.ko_raw_abundance.biom,viral_illumina_blast2tsv/V11.illumina.cat.fastq.ko_raw_abundance.biom,viral_illumina_blast2tsv/V12.illumina.cat.fastq.ko_raw_abundance.biom,viral_illumina_blast2tsv/V13.illumina.cat.fastq.ko_raw_abundance.biom -o viral_illumina_blast2tsv/VMG.illumina.ko_raw_abundance.biom

biom summarize-table -i viral_illumina_blast2tsv/VMG.illumina.ko_raw_abundance.biom  -o viral_illumina_blast2tsv/VMG.illumina.ko_raw_abundance_summarize.txt
	
biom convert -i viral_illumina_blast2tsv/VMG.illumina.ko_raw_abundance.biom -o viral_illumina_blast2tsv/VMG.illumina.ko_raw_abundance.txt --table-type="OTU table" --to-tsv
```

