---
title: "Assembly and ORF Prediction"
output:
  html_document:
    keep_md: true
---

## Viral Metagenome

Assembly the viral metagenome data using Spades.

```{r, engine='bash'}
cat viral_qc/*_finalQC.fastq > VMG.cat.fastq

python SPAdes-3.5.0-Linux/bin/spades.py -k 21,33,55,77,99,127 --only-assembler --sc -s VMG.cat.fastq -m 60000 -t 2 -o vmg_cat_sc
```

Predict ORFs with prodigal from contigs >=300 bp.

```{r, engine='bash', results='hide'}
mv vmg_cat_sc/contigs.fasta vmg.contigs.fasta

prinseq-lite.pl -fasta vmg.contigs.fasta -min_len 300 -out_format 1 -out_good vmg.contigs.filter

prodigal -i vmg.contigs.filter.fasta -d vmg.orfs_nt.fasta -a vmg.orfs_aa.fasta -p meta 
```

Filter the ORFs based on length of nt, then use those sequences as a guide to remove the short seqeunces from aa file.

```{r, engine='bash', results='hide'}
sed ':a;N;/^>/M!s/\n//;ta;P;D' vmg.orfs_aa.fasta > vmg.orfs_aa.fix.fasta

awk '!/^>/ { next } { getline seq } length(seq) >= 60 { print $0 "\n" seq }' vmg.orfs_aa.fix.fasta > vmg.orfs_aa.filter.fasta

filter_fasta.py -f vmg.orfs_nt.fasta -a vmg.orfs_aa.filter.fasta -o vmg.orfs_nt.filter.fasta
```

## Microbial Metagenome

Assemble the microbial metagenome data using Spades.

```{r, engine='bash', results='hide'}
cat microbial_qc/*_finalQC.fastq > BMG.cat.fastq

python SPAdes-3.5.0-Linux/bin/spades.py -k 21,33,55,77,99,127 --only-assembler --sc -s BMG.cat.fastq -m 200000 -t 2 -o bmg_cat_sc
```

Predict ORFs with prodigal from contigs >=300 bp.

```{r, engine='bash', results='hide'}
mv bmg_cat_sc/contigs.fasta bmg.contigs.fasta

prinseq-lite.pl -fasta bmg.contigs.fasta -min_len 300 -out_format 1 -out_good bmg.contigs.filter

prodigal -i bmg.contigs.filter.fasta -d bmg.orfs_nt.fasta -a bmg.orfs_aa.fasta -p meta 
```

Filter the ORFs based on length of nt, then use those sequences as a guide to remove the short seqeunces from aa file.

```{r, engine='bash', results='hide'}
sed ':a;N;/^>/M!s/\n//;ta;P;D' bmg.orfs_aa.fasta > bmg.orfs_aa.fix.fasta

awk '!/^>/ { next } { getline seq } length(seq) >= 60 { print $0 "\n" seq }' bmg.orfs_aa.fix.fasta > bmg.orfs_aa.filter.fasta

filter_fasta.py -f bmg.orfs_nt.fasta -a bmg.orfs_aa.filter.fasta -o bmg.orfs_nt.filter.fasta
```

## Deep Viral Metagenome

Assembly the deep viral metagenome data using Spades.

```{r, engine='bash', results='hide'}
cat deep_viral_qc/*illumina_finalQC.fastq > VMG.illumina.cat.fastq

python SPAdes-3.5.0-Linux/bin/spades.py -k 21,33,55,77,99,127 --sc -s VMG.illumina.cat.fastq -m 250000 -t 2 -o vmg_illumina_cat_sc
```

Predict ORFs with prodigal from contigs >=300 bp.

```{r, engine='bash', results='hide'}
mv vmg_illumina_cat_sc/contigs.fasta vmg.illumina.contigs.fasta

prinseq-lite.pl -fasta vmg.illumina.contigs.fasta -min_len 300 -out_format 1 -out_good vmg.illumina.contigs.filter

prodigal -i vmg.illumina.contigs.filter.fasta -d vmg.illumina.orfs_nt.fasta -a vmg.illumina.orfs_aa.fasta -p meta 
```

Filter the ORFs based on length of nt, then use those sequences as a guide to remove the short seqeunces from aa file.

```{r, engine='bash', results='hide'}
sed ':a;N;/^>/M!s/\n//;ta;P;D' vmg.illumina.orfs_aa.fasta > vmg.illumina.orfs_aa.fix.fasta

awk '!/^>/ { next } { getline seq } length(seq) >= 60 { print $0 "\n" seq }' vmg.illumina.orfs_aa.fix.fasta > vmg.illumina.orfs_aa.filter.fasta

filter_fasta.py -f vmg.illumina.orfs_nt.fasta -a vmg.illumina.orfs_aa.filter.fasta -o vmg.illumina.orfs_nt.filter.fasta
```

