---
title: "Assembly and ORF Prediction"
output:
  html_document:
    keep_md: true
---

# Assembly and ORF Prediction
Description.

## Viral Metagenome - Ion Torrent Data

Assembly the Ion Torrent viral metagenome data using Spades.

```{r, engine='bash'}
cat viral_qc/*_finalQC.fastq > VMG.cat.fastq

python SPAdes-3.5.0-Linux/bin/spades.py -k 21,33,55,77,99,127 --only-assembler --sc -s VMG.cat.fastq -m 60000 -t 2 -o vmg_cat_sc
```

Predict ORFs with prodigal from contigs >=300 bp.

```{r, engine='bash'}
mv vmg_cat_sc/contigs.fasta ./

mv contigs.fasta vmg.contigs.fasta

perl prinseq-lite-0.20.4/./prinseq-lite.pl -fasta vmg.contigs.fasta -min_len 300 -out_format 1 -out_good vmg.contigs.filter

./prodigal -i vmg.contigs.filter.fasta -d vmg.orfs_nt.fasta -a vmg.orfs_aa.fasta -p meta 
```

Filter the ORFs based on length of nt, then use those sequences as a guide to remove the short seqeunces from aa file.

```{r, engine='bash'}
perl prinseq-lite-0.20.4/./prinseq-lite.pl -fasta vmg.orfs_nt.fasta -min_len 300 -out_format 1 -out_good vmg.orfs_nt.filter

filter_fasta.py -f vmg.orfs_aa.fasta -a vmg.orfs_nt.filter.fasta -o vmg.orfs_aa.filter.fasta
```

## Microbial Metagenome - Ion Torrent Data

Assembly the Ion Torrent total metagenome data using Spades.

```{r, engine='bash'}
cat microbial_qc/*.fastq > BMG.cat.fastq

python SPAdes-3.5.0-Linux/bin/spades.py -k 21,33,55,77,99,127 --only-assembler --sc -s BMG.cat.fastq -m 200000 -t 2 -o bmg_cat_sc
```

Predict ORFs with prodigal from contigs >=300 bp.

```{r, engine='bash'}
mv bmg_cat_sc/contigs.fasta ./
mv contigs.fasta bmg.contigs.fasta

perl prinseq-lite-0.20.4/./prinseq-lite.pl -fasta bmg.contigs.fasta -min_len 300 -out_format 1 -out_good bmg.contigs.filter

./prodigal -i bmg.contigs.filter.fasta -d bmg.orfs_nt.fasta -a bmg.orfs_aa.fasta -p meta 
```

Filter the ORFs based on length of nt, then use those sequences as a guide to remove the short seqeunces from aa file.

```{r, engine='bash'}
perl prinseq-lite-0.20.4/./prinseq-lite.pl -fasta bmg.orfs_nt.fasta -min_len 300 -out_format 1 -out_good bmg.orfs_nt.filter

filter_fasta.py -f bmg.orfs_aa.fasta -a bmg.orfs_nt.filter.fasta -o bmg.orfs_aa.filter.fasta
```

## Viral Metagenome - Illumina MiSeq Data

Assembly the Illumina MiSeq viral metagenome data using Spades.

```{r, engine='bash'}
cat deep_viral_qc/*.illumina.cat.fastq > VMG.illumina.cat.fastq

python SPAdes-3.5.0-Linux/bin/spades.py -k 21,33,55,77,99,127 --sc -s VMG.illumina.cat.fastq -m 250000 -t 2 -o vmg_illumina_cat_sc
```

Predict ORFs with prodigal from contigs >=300 bp.

```{r, engine='bash'}
mv vmg_illumina_cat_sc/contigs.fasta ./

mv contigs.fasta vmg.illumina.contigs.fasta

perl prinseq-lite-0.20.4/./prinseq-lite.pl -fasta vmg.illumina.contigs.fasta -min_len 300 -out_format 1 -out_good vmg.illumina.contigs.filter

./prodigal -i vmg.illumina.contigs.filter.fasta -d vmg.illumina.orfs_nt.fasta -a vmg.illumina.orfs_aa.fasta -p meta 
```

Filter the ORFs based on length of nt, then use those sequences as a guide to remove the short seqeunces from aa file.

```{r, engine='bash'}
perl prinseq-lite-0.20.4/./prinseq-lite.pl -fasta vmg.illumina.orfs_nt.fasta -min_len 300 -out_format 1 -out_good vmg.illumina.orfs_nt.filter

filter_fasta.py -f vmg.illumina.orfs_aa.fasta -a vmg.illumina.orfs_nt.filter.fasta -o vmg.illumina.orfs_aa.filter.fasta
```
