# Generate Tables for Circos Plots - Viral Metagenome
The code below calculates how reads are shared between samples and diets to generate tables that can be used to construct the Circos plots from the manuscript with a GUI (website - directions to do so at the bottom of this notebook). Results may differ slighly from what is observed in the manuscript due to differences in subsampling events.

## Sample-Sample Comparisons

```{r, engine='bash'}
cd prinseq_output/

for f in *_finalQC.fna
do
  filename=$(basename "$f")
  filename="${filename%_*}"
  mothur "#sub.sample(fasta=$f, size=100000)"
done

rm VMG_prinseq_bad_finalQC.subsample.fna

for *subsample.fna
do
    filename=$(basename "$f")
    filename="${filename%.*}"
    load-into-counting.py -k 20 -N 4 -x 8e9 --report-total-kmers -s tsv "$filename""_subsample_k20.kh" $f
done

ls *subsample_k20.kh | awk '{ ORS=" "; print; }' > config_sub.txt
printf "\n" >> config_sub.txt
ls *finalQC.subsample.fna | awk '{ ORS=" "; print; }' >> config_sub.txt
printf "\n" >> config_sub.txt
printf "50000000" >> config_sub.txt

python ../get_comb_multi_old_median_kmer.py config_sub.txt

printf "\n27CDS 55CS 55CS 27CDS Corn 40MDGS Corn Corn 40MDGS 27CDS 55CS 40MDGS 40MDGS Corn 55CS" >> config_sub.txt
printf "\n259 346 3244 222 3257 259 346 3244 222 3257 259 346 3244 222 3257" >> config_sub.txt

cd ..
```

Following script calcualtes the median shared k-mer content between each pair of samples. The script only considers reads that are shared with at least one other sample to avoid potential impact of sequencing errors on the interpretation.

```{r, engine=bash}
#wget script
python circos_make_table.py -d prinseq_output/ -c prinseq_output/config_sub.txt -i T -s subsample.fna.comb > circos_list_sub.txt
```

```{r}
options(stringsAsFactors=FALSE)
matrix_out <- data.frame(matrix(NA, nrow = 15, ncol = 15))
pw <- read.table("circos_list_sub.txt", sep="\t", header=FALSE)
r_c_names <- unique(pw$V1)
row.names(matrix_out) <- r_c_names
colnames(matrix_out) <- r_c_names

for(i in 1:nrow(pw)) {
	matrix_out[pw$V1[i], pw$V2[i]] = pw$V3[i]
}

write.table(matrix_out, file="circos_table_sub.txt", sep="\t", row.names=TRUE, col.names=NA, quote=FALSE)
```

```{r, engine=bash}
sed -E $'s/\t40MDGS-259/labels\t40MDGS-259/g' circos_table_sub.txt > circos_table2_sub.txt

printf "labels\n0,159,107\n255,211,0\n196,2,51\n196,2,51\n196,2,51\n255,211,0\n0,159,107\n0,159,107\n0,135,189\n0,135,189\n0,159,107\n255,211,0\n0,135,189\n255,211,0\n196,2,51" > circos_row_colors.txt
paste circos_row_colors.txt circos_table2_sub.txt > circos_viral_all_samples_upload.txt

printf "labels\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n0,135,189\n0,135,189\n211,211,211\n211,211,211\n0,135,189\n211,211,211\n211,211,211" > circos_row_colors_27cds.txt
paste circos_row_colors_27cds.txt circos_table2_sub.txt > circos_viral_27cds_upload.txt

printf "labels\n211,211,211\n211,211,211\n196,2,51\n196,2,51\n196,2,51\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n196,2,51" > circos_row_colors_55cs.txt
paste circos_row_colors_55cs.txt circos_table2_sub.txt > circos_viral_55cs_upload.txt

printf "labels\n0,159,107\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n0,159,107\n0,159,107\n211,211,211\n211,211,211\n0,159,107\n211,211,211\n211,211,211\n211,211,211\n211,211,211" > circos_row_colors_40mdgs.txt
paste circos_row_colors_40mdgs.txt circos_table2_sub.txt > circos_viral_40mdgs_upload.txt

printf "labels\n211,211,211\n255,211,0\n211,211,211\n211,211,211\n211,211,211\n255,211,0\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n255,211,0\n211,211,211\n255,211,0\n211,211,211" > circos_row_colors_corn.txt
paste circos_row_colors_corn.txt circos_table2_sub.txt > circos_viral_corn_upload.txt
```
