---
title: "microbial Metagenome Circos Plots"
output:
  html_document:
    keep_md: true
---

The code below calculates how reads are shared between samples and diets to generate tables that can be used to construct the Circos plots from the manuscript with a GUI (website - directions to do so at the bottom of this notebook). Results may differ slighly from what is observed in the manuscript due to differences in subsampling events.

## Sample-Sample Comparisons

```{r, engine='bash', results='hide'}
cd microbial_qc/
 
for f in *finalQC.fna
do
  filename=$(basename "$f")
  filename="${filename%.*}"
  mothur "#sub.sample(fasta=$f, size=450000)"
done

for f in *subsample.fna
do
    filename=$(basename "$f")
    filename="${filename%.*}"
    load-into-counting.py -k 20 -N 4 -x 1e9 --report-microbial-kmers -s tsv "$filename""_subsample_k20.kh" $f
done

ls *subsample_k20.kh | awk '{ ORS=" "; print; }' > config_sub.txt
printf "\n" >> config_sub.txt
ls *subsample.fna | awk '{ ORS=" "; print; }' >> config_sub.txt
printf "\n" >> config_sub.txt
printf "55000000" >> config_sub.txt
 
python ../scripts/get_comb_multi_old_median_kmer.py config_sub.txt
 
printf "\n27CDS 55CS 55CS 27CDS Corn 40MDGS 27CDS 27CDS 55CS 40MDGS Corn Corn Corn 40MDGS 27CDS 55CS 40MDGS 40MDGS Corn 55CS" >> config_sub.txt
printf "\n259 346 3244 222 3257 259 346 3244 222 3257 346 259 3244 222 3257 259 346 3244 222 3257" >> config_sub.txt
 
cd ..
```

Following script calcualtes the median shared k-mer content between each pair of samples. The script only considers reads that are shared with at least one other sample to avoid potential impact of sequencing errors on the interpretation. Shared is defined as a median k-mer count >1.

```{r, engine='bash'}
perl scripts/khmer_multi_threshold.pl -khmer_multi_dir=qc_microbial -threshold=1

mv *.comb.keep.txt microbial_qc/

python scripts/circos_make_table.py -d microbial_qc/ -c microbial_qc/config_sub.txt -i T -s subsample.fna.comb.keep.txt > microbial_circos_list_sub.txt
```

```{r, cache=FALSE}
options(stringsAsFactors=FALSE)
matrix_out <- data.frame(matrix(NA, nrow = 20, ncol = 20))
pw <- read.table("microbial_circos_list_sub.txt", sep="\t", header=FALSE)
r_c_names <- unique(pw$V1)
row.names(matrix_out) <- r_c_names
colnames(matrix_out) <- r_c_names

for(i in 1:nrow(pw)) {
	matrix_out[pw$V1[i], pw$V2[i]] = pw$V3[i]
}

write.table(matrix_out, file="microbial_circos_table_sub.txt", sep="\t", row.names=TRUE, col.names=NA, quote=FALSE)
```

```{r, engine='bash'}
sed -E $'s/\tCorn-3257/labels\tCorn-3257/g' microbial_circos_table_sub.txt > microbial_circos_table2_sub.txt

printf "labels\n255,211,0\n0,159,107\n196,2,51\n196,2,51\n0,135,189\n196,2,51\n0,135,189\n0,159,107\n0,135,189\n255,211,0\n255,211,0\n0,135,189\n196,2,51\n0,159,107\n0,135,189\n0,159,107\n255,211,0\n0,159,107\n196,2,51\n255,211,0" > microbial_circos_row_colors.txt
paste microbial_circos_row_colors.txt microbial_circos_table2_sub.txt > circos_microbial_all_samples_upload.txt

printf "labels\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n0,135,189\n211,211,211\n0,135,189\n211,211,211\n0,135,189\n211,211,211\n211,211,211\n0,135,189\n211,211,211\n211,211,211\n0,135,189\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n211,211,211" > microbial_circos_row_colors_27cds.txt
paste microbial_circos_row_colors_27cds.txt microbial_circos_table2_sub.txt > circos_microbial_27cds_upload.txt

printf "labels\n211,211,211\n211,211,211\n196,2,51\n196,2,51\n211,211,211\n196,2,51\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n196,2,51\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n196,2,51\n211,211,211" > microbial_circos_row_colors_55cs.txt
paste microbial_circos_row_colors_55cs.txt microbial_circos_table2_sub.txt > circos_microbial_55cs_upload.txt

printf "labels\n211,211,211\n0,159,107\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n0,159,107\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n0,159,107\n211,211,211\n0,159,107\n211,211,211\n0,159,107\n211,211,211\n211,211,211" > microbial_circos_row_colors_40mdgs.txt
paste microbial_circos_row_colors_40mdgs.txt microbial_circos_table2_sub.txt > circos_microbial_40mdgs_upload.txt

printf "labels\n255,211,0\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n255,211,0\n255,211,0\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n211,211,211\n255,211,0\n211,211,211\n211,211,211\n255,211,0" > microbial_circos_row_colors_corn.txt
paste microbial_circos_row_colors_corn.txt microbial_circos_table2_sub.txt > circos_microbial_corn_upload.txt
```

## Diet-Diet Comparisons
```{r, engine='bash', results='hide'}
cd microbial_qc/
mkdir sub_sample_comb
mv *.comb.keep.txt sub_sample_comb
mv *subsample.fna.comb sub_sample_comb

cat BMG.4_cd454.subsample.fna BMG.10_cd454.subsample.fna BMG.13_cd454.subsample.fna BMG.16_cd454.subsample.fna BMG.17_cd454.subsample.fna > 27cds_sub_cat.fna

cat BMG.14_cd454.subsample.fna BMG.1_cd454.subsample.fna BMG.2_cd454.subsample.fna BMG.8_cd454.subsample.fna BMG.20_cd454.subsample.fna > corn_sub_cat.fna

cat BMG.15_cd454.subsample.fna BMG.3_cd454.subsample.fna BMG.6_cd454.subsample.fna BMG.7_cd454.subsample.fna BMG.19_cd454.subsample.fna > 40mdgs_sub_cat.fna

cat BMG.5_cd454.subsample.fna BMG.11_cd454.subsample.fna BMG.12_cd454.subsample.fna BMG.9_cd454.subsample.fna BMG.18_cd454.subsample.fna > 55cs_sub_cat.fna

for f in *sub_cat.fna
do
    filename=$(basename "$f")
    filename="${filename%.*}"
    load-into-counting.py -k 20 -N 4 -x 1e9 --report-total-kmers -s tsv "$filename""_sub_cat_k20.kh" $f
done

ls *sub_cat_k20.kh | awk '{ ORS=" "; print; }' > config_cat_sub.txt
printf "\n" >> config_cat_sub.txt
ls *sub_cat.fna | awk '{ ORS=" "; print; }' >> config_cat_sub.txt
printf "\n" >> config_cat_sub.txt
printf "50000000" >> config_cat_sub.txt

python ../scripts/get_comb_multi_old_median_kmer.py config_cat_sub.txt

printf "\n27CDS 40MDGS 55CS Corn" >> config_cat_sub.txt

cd ..
```

```{r, engine='bash'}
perl scripts/khmer_multi_threshold.pl -khmer_multi_dir=qc_microbial -threshold=1

mv *.comb.keep.txt qc_microbial/

python scripts/circos_make_table.py -d qc_microbial/ -c qc_microbial/config_cat_sub.txt -i F -s sub_cat.fna.comb.keep.txt > microbial_circos_list_sub_cat.txt

mkdir qc_microbial/sub_diet_comb
mv qc_microbial/*sub_cat.fna.comb.keep.txt qc_microbial/sub_diet_comb
mv qc_microbial/*sub_cat.fna.comb qc_microbial/sub_diet_comb
```

```{r, cache=FALSE}
options(stringsAsFactors=FALSE)
matrix_out <- data.frame(matrix(NA, nrow = 4, ncol = 4))
pw <- read.table("microbial_circos_list_sub_cat.txt", sep="\t", header=FALSE)
r_c_names <- unique(pw$V1)
row.names(matrix_out) <- r_c_names
colnames(matrix_out) <- r_c_names

for(i in 1:nrow(pw)) {
	matrix_out[pw$V1[i], pw$V2[i]] = pw$V3[i]
}

write.table(matrix_out, file="microbial_circos_table_sub_cat.txt", sep="\t", row.names=TRUE, col.names=NA, quote=FALSE)
```

```{r, engine='bash'}
sed -E $'s/\t40MDGS/labels\t40MDGS/g' microbial_circos_table_sub_cat.txt > microbial_circos_table_sub_cat2.txt

printf "labels\n0,159,107\n196,2,51\n0,135,189\n255,211,0" > microbial_circos_row_colors_cat.txt

paste microbial_circos_row_colors_cat.txt microbial_circos_table_sub_cat2.txt > circos_microbial_diets_upload.txt
```

## Circos GUI

GUI: http://mkweb.bcgsc.ca/tableviewer/visualize/

Go to the settings tab at the top of the page and change the following (only needs to be done once):

- Under labels, segment fold = bold
- Under cell ribbons, ribbon order = cell size, descending and ribbon layer order = large on top of small
- Under contribution tracks, check hide
- Under image format, check hide relative tick marks, hide absolute tick marks, hide first tick label

Click save. Switch to the visualize tab.

Six files for upload:

- circos_microbial_all_samples_upload.txt
- circos_microbial_diets_upload.txt

Under section 2A, upload each file and check "col with row colors".

Download the results.

Click the visualize tab again to upload another file.