---
title: "Partial Least Squares Regression"
output:
  html_document:
    keep_md: true
---

# Partial Least Squares Regression - Ion Torrent Viral Metagenomes

Investigating multivariate linear relationships between rlog tranformed KO annotations in the viral metagenome and dietary components. This is followed up by a univariate multiple linear regressions (MLR) for each KO to confirm the potetnial relationships identified by PLSR.

```{r,}
library(pls)
library(phyloseq)
library(DESeq2)
library(vsn)
library(KEGGREST)
library(car)
library(parallel)
library(QuantPsyc)
library(reshape2)
options(stringsAsFactors=FALSE)

table <- read.table("viral_blast2tsv_filter2/VMG.ko_raw_abundances.txt", header=TRUE, comment.char="", skip=1, row.names=1, sep="\t")
table_matrix <- as.matrix(table)
map <- read.table("viral_mapping_ko.txt", sep="\t", header=TRUE, row.names=1)
OTU = otu_table(table_matrix, taxa_are_rows = TRUE)
SAMPLEDATA = sample_data(map)
physeq = phyloseq(OTU, SAMPLEDATA)
dds = phyloseq_to_deseq2(physeq, ~Diet)

rld <- rlog(dds, blind = FALSE)
rlogmat <- assay(rld)
rlogdf <- as.data.frame(rlogmat)
write.table(rlogdf, file="viral_blast2tsv_filter2/VMG.ko_rlog.txt", col.names=NA, row.names=TRUE, quote=FALSE, sep="\t")
rlogmat_t <- t(rlogmat)
map2 <- as.matrix(map[c("TDN", "Zn", "Protein")])
pls_df <- data.frame(y=I(rlogmat_t), x=I(map2))
pls.options(parallel = 4)
pls_result <- plsr(y ~ x, data = pls_df, validation = "LOO", scale=TRUE)
capture.output(summary(pls_result, what="training"), file="plsr_result.txt")
summaryfile <- read.table("plsr_result.txt", skip=7)
summaryfile$C1_exp <- summaryfile$V2
summaryfile$C2_exp <- (summaryfile$V3 - summaryfile$V2)
summaryfile$C3_exp <- (summaryfile$V4 - summaryfile$V3)
c1_top <- summaryfile[ which(summaryfile$C1_exp > 30), ]
dim(c1_top)
c2_top <- summaryfile[ which(summaryfile$C2_exp > 30), ]
dim(c2_top)
c3_top <- summaryfile[ which(summaryfile$C3_exp > 30), ]
dim(c3_top)
pls_top <- rbind(c1_top,c2_top,c3_top)
pls_top_list <- unique(pls_top$V1) #101
```

MLR on KOs that have >= 30% variation explained by any component.

```{r,}
rlog_psl <- rlogdf[pls_top_list,]
rlog_psl_t <- t(rlog_psl)


res_tdn <- data.frame(id = character(0), tdn_radj=character(0), tdn_p_val=character(0), tdn_coeff=character(0), tdn_coeff_stand=character(0))
for (i in 1:ncol(rlog_psl_t)) {
	mlr <- lm(rlog_psl_t[,i] ~ TDN, data=map)
	summ <- summary(mlr)
	tdn_p_val <- summ$coefficients[8]
	tdn_radj <-  summ$adj.r.squared
	name <- colnames(rlog_psl_t)[i]
	tdn_coeff <- mlr$coefficients[2]
	tdn_coeff_stand <- lm.beta(mlr)[1]
	res_tdn[i, ] <- c(name, tdn_radj, tdn_p_val, tdn_coeff, tdn_coeff_stand)
}

tdn_sig <- subset(res_tdn, tdn_p_val < 0.05)
dim(tdn_sig) #71


res_zn <- data.frame(id = character(0), zn_radj=character(0), zn_p_val=character(0), zn_coeff=character(0), zn_coeff_stand=character(0))
for (i in 1:ncol(rlog_psl_t)) {
	mlr <- lm(rlog_psl_t[,i] ~ Zn, data=map)
	summ <- summary(mlr)
	zn_p_val <- summ$coefficients[8]
	zn_radj <-  summ$adj.r.squared
	name <- colnames(rlog_psl_t)[i]
	zn_coeff <- mlr$coefficients[2]
	zn_coeff_stand <- lm.beta(mlr)[1]
	res_zn[i, ] <- c(name, zn_radj, zn_p_val, zn_coeff, zn_coeff_stand)
}

zn_sig <- subset(res_zn, zn_p_val < 0.05)
dim(zn_sig) #14


res_protein <- data.frame(id = character(0), protein_radj=character(0), protein_p_val=character(0), protein_coeff=character(0), protein_coeff_stand=character(0))
for (i in 1:ncol(rlog_psl_t)) {
	mlr <- lm(rlog_psl_t[,i] ~ Protein, data=map)
	summ <- summary(mlr)
	protein_p_val <- summ$coefficients[8]
	protein_radj <-  summ$adj.r.squared
	name <- colnames(rlog_psl_t)[i]
	protein_coeff <- mlr$coefficients[2]
	protein_coeff_stand <- lm.beta(mlr)[1]
	res_protein[i, ] <- c(name, protein_radj, protein_p_val, protein_coeff, protein_coeff_stand)
}

protein_sig <- subset(res_protein, protein_p_val < 0.05)
dim(protein_sig) #4


length(intersect(tdn_sig$id, zn_sig$id)) #1
length(intersect(tdn_sig$id, protein_sig$id)) #2
length(intersect(protein_sig$id, zn_sig$id)) #0

```

Heatmap of MLR results.

```{r}
library(RCurl)
library(R.utils)
library(pheatmap)

res1 <- merge(res_tdn, res_zn, by="id")
final_pls_table <- merge(res1, res_protein, by="id")
reg_sign <- subset(final_pls_table, tdn_p_val < 0.05 | zn_p_val < 0.05 | protein_p_val < 0.05)



reg_hm <- data.frame(id = character(0), TDN=numeric(0), Zn=numeric(0), Protein=numeric(0))
options(stringsAsFactors=FALSE)
for (i in 1:nrow(reg_sign)) {
	id <- reg_sign[i, "id"]

	tdn_p <- reg_sign[i,"tdn_p_val"]
	if (tdn_p < 0.05) {
		tdn <- reg_sign[i,"tdn_coeff_stand"]
	} else {
		tdn <- 0
	}

	zn_p <- reg_sign[i,"zn_p_val"]
	if (zn_p < 0.05) {
		zn <- reg_sign[i,"zn_coeff_stand"]
	} else {
		zn <- 0
	}
	
	protein_p <- reg_sign[i,"protein_p_val"]
	if (protein_p < 0.05) {
		protein <- reg_sign[i,"protein_coeff_stand"]
	} else {
		protein <- 0
	}
	
	reg_hm[i, ] <- c(id, tdn, zn, protein)
}

reg_hm$TDN <- as.numeric(reg_hm$TDN)
reg_hm$Zn <- as.numeric(reg_hm$Zn)
reg_hm$Protein <- as.numeric(reg_hm$Protein)
is.num <- sapply(reg_hm, is.numeric)
reg_hm[is.num] <- lapply(reg_hm[is.num], round, 3)

enz_list <- reg_hm$id
enz_names <- data.frame(ID = character(0), Name = character(0), Def=character(0))	
options(stringsAsFactors=FALSE)
for (i in 1:length(enz_list)) {
	enz <- enz_list[i]
	enz2 <- noquote(enz)
	url_get <- noquote(paste0("http://togows.dbcls.jp/entry/orthology/",enz2,"/name"))
	names <- getURL(url_get)
	Sys.sleep(1) 
	names <- gsub("[\r\n]", "", names)
	url_get2 <- noquote(paste0("http://togows.dbcls.jp/entry/orthology/",enz2,"/definition"))
	defs <- getURL(url_get2)
	Sys.sleep(1) 
	defs <- gsub("[\r\n]", "", defs)	
	enz_names[i, ] <- c(enz, names, defs)
}

colnames(enz_names)[1] <- "id"

enz_names2 <- enz_names
enz_names2$Def2 <- gsub(" \\[EC(.)+", "", enz_names2$Def)
enz_names2$Def3 <- capitalize(enz_names2$Def2)
reg_hm2 <- merge(reg_hm, enz_names2, by="id")
rownames(reg_hm) <- reg_hm$id
reg_hm <- reg_hm[c(-1)]
reg_hm_mat <- as.matrix(reg_hm)

reg_hm3 <- reg_hm2[c("id", "Def", "TDN", "Zn", "Protein")]
write.table(reg_hm3, file="plsr_mlr_hm_data.txt", row.names=FALSE, quote=FALSE, sep="\t")

reg_hm_sort <- reg_hm3[order(reg_hm3$TDN),]
write.table(reg_hm_sort, file="plsr_mlr_hm_data_sort.txt", row.names=FALSE, quote=FALSE, sep="\t")


bk <- c(-1.500, -1.125, -0.750, -0.375, -0.0001, 0.0001, 0.375, 0.750, 1.125, 1.500)  
col_pal <- c("#b2182b", "#d6604d", "#f4a582", "#fddbc7", "#f7f7f7", "#d1e5f0", "#92c5de", "#4393c3", "#2166ac")

tiff("figures/figure_s3_nolabel.tiff", height=8, width=6, res=600, units = 'in', compression=c("lzw"))
pheatmap(reg_hm_mat, 
	cluster_row = F,
    cluster_cols = F,
    color = col_pal,
    fontsize = 6.5,
    fontsize_row=6, 
    fontsize_col = 12,
    labels_row = reg_hm2$Def3,
    border_color = "black",
    breaks=bk)
dev.off()
```
