---
title: "Quality Control Viral Metagenomes"
output:
  html_document:
    keep_md: true
---

After seqeuncing the Torrent Server software demulitplexed samples, trimed off adaptors and barcodes, and removed reads less than 100 basepairs. More precisley the following commands were used within the Torrent Server:

--barcode-mode 1 --barcode-cutoff 0 --min-read-length 100 --trim-min-read-len 100

The resulting FASTQ files were further filtered below.

Explain prep and issues with it. Also diff library sizes, so some different filters were applied to samples from the same dataset in regards to that.

The file transp_adapt_remove.cat.txt was supplied to direct removal of adaptors, barcodes, and transposon seqeunces. Further, observed GC and k-mer bias was noted in the 5' and 3' ends.  In order to begin to alleviate those biases, the first 20 bp of the read was trimmed and a sample dependent trimming from the 3' end was done (more 3' trimming later).

Not using a loop beacuse outputting names that match mapping files and code used in the analysis. These names are not congruent with the SRA file names. Also some files require different quality filtering due to run specific quality patterns.

```{r, engine='bash', results='hide'}
mkdir viral_qc
chmod 775 Trimmomatic-0.33/*

java -jar Trimmomatic-0.33/trimmomatic-0.33.jar SE -phred33 raw_viral/V1.fastq viral_qc/V1_trimm.fastq ILLUMINACLIP:Trimmomatic-0.33/transp_adapt_remove.cat.txt:2:20:10 CROP:175 MINLEN:75

java -jar Trimmomatic-0.33/trimmomatic-0.33.jar SE -phred33 raw_viral/V2.fastq viral_qc/V2_trimm.fastq ILLUMINACLIP:Trimmomatic-0.33/transp_adapt_remove.cat.txt:2:20:10 CROP:175 MINLEN:75

java -jar Trimmomatic-0.33/trimmomatic-0.33.jar SE -phred33 raw_viral/V3.fastq viral_qc/V3_trimm.fastq ILLUMINACLIP:Trimmomatic-0.33/transp_adapt_remove.cat.txt:2:20:10 CROP:175 MINLEN:75

java -jar Trimmomatic-0.33/trimmomatic-0.33.jar SE -phred33 raw_viral/V4.fastq viral_qc/V4_trimm.fastq ILLUMINACLIP:Trimmomatic-0.33/transp_adapt_remove.cat.txt:2:20:10 CROP:175 MINLEN:75

java -jar Trimmomatic-0.33/trimmomatic-0.33.jar SE -phred33 raw_viral/V5.fastq viral_qc/V5_trimm.fastq ILLUMINACLIP:Trimmomatic-0.33/transp_adapt_remove.cat.txt:2:20:10 CROP:100 MINLEN:75

java -jar Trimmomatic-0.33/trimmomatic-0.33.jar SE -phred33 raw_viral/V6.fastq viral_qc/V6_trimm.fastq ILLUMINACLIP:Trimmomatic-0.33/transp_adapt_remove.cat.txt:2:20:10 CROP:175 MINLEN:75

java -jar Trimmomatic-0.33/trimmomatic-0.33.jar SE -phred33 raw_viral/V7.fastq viral_qc/V7_trimm.fastq ILLUMINACLIP:Trimmomatic-0.33/transp_adapt_remove.cat.txt:2:20:10 CROP:175 MINLEN:75

java -jar Trimmomatic-0.33/trimmomatic-0.33.jar SE -phred33 raw_viral/V8.fastq viral_qc/V8_trimm.fastq ILLUMINACLIP:Trimmomatic-0.33/transp_adapt_remove.cat.txt:2:20:10 CROP:175 MINLEN:75

java -jar Trimmomatic-0.33/trimmomatic-0.33.jar SE -phred33 raw_viral/V9.fastq viral_qc/V9_trimm.fastq ILLUMINACLIP:Trimmomatic-0.33/transp_adapt_remove.cat.txt:2:20:10 CROP:175 MINLEN:75

java -jar Trimmomatic-0.33/trimmomatic-0.33.jar SE -phred33 raw_viral/V10.fastq viral_qc/V10_trimm.fastq ILLUMINACLIP:Trimmomatic-0.33/transp_adapt_remove.cat.txt:2:20:10 CROP:175 MINLEN:75

java -jar Trimmomatic-0.33/trimmomatic-0.33.jar SE -phred33 raw_viral/V11.fastq viral_qc/V11_trimm.fastq ILLUMINACLIP:Trimmomatic-0.33/transp_adapt_remove.cat.txt:2:20:10 CROP:175 MINLEN:75

java -jar Trimmomatic-0.33/trimmomatic-0.33.jar SE -phred33 raw_viral/V12.fastq viral_qc/V12_trimm.fastq ILLUMINACLIP:transp_adapt_remove.cat.txt:2:20:10 CROP:175 MINLEN:75

java -jar Trimmomatic-0.33/trimmomatic-0.33.jar SE -phred33 raw_viral/V13.fastq viral_qc/V13_trimm.fastq ILLUMINACLIP:transp_adapt_remove.cat.txt:2:20:10 CROP:175 MINLEN:75

java -jar Trimmomatic-0.33/trimmomatic-0.33.jar SE -phred33 raw_viral/V14.fastq viral_qc/V14_trimm.fastq ILLUMINACLIP:Trimmomatic-0.33/transp_adapt_remove.cat.txt:2:20:10 CROP:175 MINLEN:75

java -jar Trimmomatic-0.33/trimmomatic-0.33.jar SE -phred33 raw_viral/V15.fastq viral_qc/V15_trimm.fastq ILLUMINACLIP:Trimmomatic-0.33/transp_adapt_remove.cat.txt:2:20:10 CROP:175 MINLEN:75
```

To remove artifical duplicates we use cd-hit-454.  We noticed early on the transposon library preps with the Ion Torrent Museek kits introduces (unfortunately sometimes many) duplicates (associated with viral sample?). While the software was originally designed for 454 sequencing, Ion Torrent shares many of the same error profiles as 454 and cd-hit-454 appears to work fairly well for removing duplicates in our data.

```{r, engine='bash', results='hide'}
export OMP_NUM_THREADS=10

for f in viral_qc/*.fastq
do
  filename=$(basename "$f")
  filename="${filename%_*}"
  cd-hit-454 -i $f -o viral_qc/"$filename""_cd454.fastq" -M 6100 -T 10
done
```

Now, we wanted to check for other types of duplicates and some that may have been missed by cd-hit-454. Further, we can use prinseq to remove transposon associated seqeunces that kept showing up in the 3' end.

```{r, engine='bash', results='hide'}
for f in viral_qc/*_cd454.fastq
do
  filename=$(basename "$f")
  filename="${filename%_*}"
  prinseq-lite.pl -custom_params "TGAACTG 1;GAACTGA 1;AACTGAC 1;ACTGACG 1;CTGACGC 1;TGACGCA 1;GACGCAC 1;ACGCACG 1;CGCACGA 1;GCACGAA 1;" -derep 14 -lc_method dust -lc_threshold 7 -fastq $f -out_format 3 -out_good viral_qc/"$filename""_prinseq"
done
```

Use prefix length to remove duplicates with exact matches over first 75 basepairs.

```{r, engine='bash', results='hide'}
for f in viral_qc/*_prinseq.fastq
do
  filename=$(basename "$f")
  filename="${filename%_*}"
  prinseq-lite.pl -trim_to_len 75 -derep 1 -fastq $f -out_format 2 -out_good viral_qc/"$filename""_truncatederep"
done
```

Use qiime to remove those identified as prefix duplicates and khmer to get a FASTA version.

```{r, engine='bash', results='hide'}
for f in viral_qc/*_truncatederep.fasta
do
  filename=$(basename "$f")
  filename="${filename%_*}"
  grep ">" $f | cut -c 2- > viral_qc/"$filename"_keep_ids.txt
  filter_fasta.py -f viral_qc/"$filename"_prinseq.fastq -s viral_qc/"$filename"_keep_ids.txt -o viral_qc/"$filename"_finalQC.fastq
  fastq-to-fasta.py viral_qc/"$filename"_finalQC.fastq -o viral_qc/"$filename"_finalQC.fna
done
```

Read counts.

```{r, engine='bash'}
grep -c ">" viral_qc/*finalQC.fna
```

Look at rRNA contamination rates. For some reason I can only get this to run from within the examples folder.  So, move all the fasta files generated to the examples folder within rRNA predictor.

```{r, engine='bash', results='hide'}
mkdir rRNA_prediction/examples/e1/viral_input
mkdir rRNA_prediction/examples/e1/viral_output
cp viral_qc/*.fna rRNA_prediction/examples/e1/viral_input/
RRNA=`cd rRNA_prediction/rRNA_hmm_fs_wst_v0; pwd`
export PATH=$PATH:$RRNA
cd rRNA_prediction/examples/e1
../../scripts/rRNA_hmm_run_wst_v0.pl  viral_input viral_output
cd ../../..
```

Check the number of rRNA hits for each sample using custom script to parse outputs:

```{r, engine='bash'}
for f in rRNA_prediction/examples/e1/viral_output/*.coord
do
  echo $f
  perl scripts/parse_rRNA_output.pl -rrna $f
done
```